tools/build-image-if-needed.sh

```diff
@@ -15,5 +15,6 @@ if git diff --exit-code remotes/origin/master...HEAD -- ${BUILD_DIRECTORY}; then
   echo "${BUILD_DIRECTORY} files not modified; no rebuild required"
 else
   echo "${BUILD_DIRECTORY} files modified; ${TAG_NAME} requires a rebuild"
-  docker build ${BUILD_DIRECTORY} -t ${TAG_NAME}
+  docker buildx build --platform linux/amd64 --no-cache ${BUILD_DIRECTORY} -t ${TAG_NAME}
+  docker buildx build --platform linux/arm64 --no-cache ${BUILD_DIRECTORY} -t ${TAG_NAME}
```

Comment:
Just tested again:
```
error: multiple platforms feature is currently not supported for docker driver. Please switch to a different driver (eg. "docker buildx create --use")
```
I imagine it may be better to have this run only once before all buildx commands, right? I.e., outside of this script.

Reply:
Ah, you'll need to copy these 4 lines from `.github/workflows/images.yml`:

https://github.com/PrairieLearn/PrairieLearn/blob/df37ad2fad7aa7205e3fa9640c83d92f7741f59f/.github/workflows/images.yml#L16-L19

Things should work after that - I suspect you also won't need the `--platform=$BUILDPLATFORM` in the `Dockerfile` once this is here, though that's definitely worth verifying.

Reply:
It's still running, but it seems to have done the trick... thanks.

Reply:
something's hung for a while now 😢

Reply:
I noticed something similar locally as well, a bit weird, installing `matplotlib` in arm takes more than an hour to run, not sure why. I'm testing with replacing the installation of matplotlib from pip to apt to see if it speeds up.

Reply:
It looks as though this image has pip 9.0 installed:

```
$ docker run --rm prairielearn/grader-c:latest python3 -m pip --version
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
pip 9.0.1 from /usr/lib/python3/dist-packages (python 3.8)
```

Per https://github.com/pypa/manylinux, the `manylinux2014_aarch64` wheel that matplotlib ships is only supported by pip >= 19.3. It may be trying to build matplotlib from source? I'd recommend upgrading pip to a newer version and trying again with installing matplotlib via pip + `requirements.txt`.

Reply:
Upgrading pip via pip -U did the trick, and removed the need for cython and explicit pillow. It also made the build incredibly faster.